pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -116,7 +116,7 @@ func (configgen *ConfigGeneratorImpl) BuildListeners(node *model.Proxy,
 
 	builder.patchListeners()
 	l := builder.getListeners()
-	if node.EnableHBONE() {
+	if node.EnableHBONE() && !node.IsWaypointProxy() {
```

Comment:
The reason I added this is because https://github.com/istio/istio/blob/experimental-ambient/manifests/profiles/ambient.yaml#L10 enables this HBONE flag and the `tunnel` listener is being added twice (causing error of duplicated listeners).